test/custom-parser.test.js

```diff
@@ -647,7 +647,7 @@ test('Should parse empty bodies as a string', t => {
   })
 
   fastify.route({
-    method: ['POST', 'DELETE'],
+    method: ['POST'],
```

Comment:
I would expect this to work as it was before. cc @nwoltman

Reply:
It will skip the body parsing too. May I add a new test which cheks if the body is null?

Reply:
This test is a little misleading because `body: '',` gets passed as an option to `simple-get` but `simple-get` ignores this and doesn't set the `Content-Length` header. With the new changes, `DELETE` requests are interpreted as there is no body so the body won't be parsed (whereas `POST` requests expect a body so it will still be parsed).

This test could stay mostly the same if `Content-Length: 0` were added to the `headers` passed to `sget`.